flair/data.py

```diff
@@ -198,22 +200,76 @@ def __init__(self, text: str = None, use_tokenizer: bool = False, labels: List[L
 
         self._embeddings: Dict = {}
 
-        # optionally, directly instantiate with sentence tokens
+        # if text is passed, instantiate sentence with tokens (words)
         if text is not None:
 
-            # tokenize the text first if option selected, otherwise assumes whitespace tokenized text
+            # tokenize the text first if option selected
             if use_tokenizer:
-                sentences = split_single(text)
+
+                # use segtok for tokenization
                 tokens = []
+                sentences = split_single(text)
                 for sentence in sentences:
                     contractions = split_contractions(word_tokenizer(sentence))
                     tokens.extend(contractions)
 
-                text = ' '.join(tokens)
+                # determine offsets for whitespace_after field
+                index = text.index
+                running_offset = 0
+                last_word_offset = -2
+                last_token = None
+                for word in tokens:
+                    token = Token(word)
+                    self.add_token(token)
+                    try:
+                        word_offset = index(word, running_offset)
+                    except:
+                        word_offset = last_word_offset = +1
```

Comment:
@alanakbik This line looks a bit weird, I mean the `= +1` operator here.

Reply:
@stefan-it yeah, unfortunately in some (very) rare cases the segtok tokenization produces a strange error in which it produces a token that is not actually in the sentence. this can cause the index search to fail, so just to be sure there is a fallback in such cases to default to the next position in the sentence.